### PR TITLE
Fix tree restricted search infinite wrap

### LINKED ISSUE #1332: search, keyboard stops to respond after typing 4 slashes
Typing four times a slash locks up the keyboard/application. After three slashes no additional slashes appear on screen and the keyboard appears to be locked. Music keeps playing. 

Only a `kill -9` allows me to stop cmus after. 

Up to 3 the keyboard stays responsive, the slashes appear on screen and i can remove them using backspace. Each time on the fourth the keyboard end up in a state i do not how to get out of.

---

1) `uname -a`:
```
Ubuntu 22.04, x86_64 
```

2) `cmus --version`:
```
cmus --version
cmus v2.9.1
Copyright 2004-2006 Timo Hirvonen
Copyright 2008-2016 Various Authors
```

3) `cmus --plugins`:
```
default Ubuntu apt install.
```

### PR DESCRIPTION
**Pull request by pgaskin**
fixes #1332

### CODE REVIEW

**Comment by pgaskin** [comment]
@flyingmutant, this should be rebase-merged to keep the refactor separate from the fix.

File: tree.c [hunk 1]
```diff
@@ -416,6 +416,20 @@ static int tree_search_get_current(void *data, struct iter *iter, enum search_di
...
+			 */
+			if (dir == SEARCH_FORWARD) {
+				track = to_tree_track(rb_first(&track->album->track_root));
+			} else {
+				track = to_tree_track(rb_last(&track->album->track_root));
+			}
+		}
 		tree_search_track_to_iter(track, iter);
 		return 1;
 	}
```
Comment (gavtroy) [inline tree.c:435]: For completeness there should also be a SEARCH_BACKWARD version for the "artist not expanded" case below.

Comment (pgaskin) [reply tree.c:435]: Fixed.

**Review by gavtroy — APPROVED** [review]

**Comment by flyingmutant** [comment]
> @flyingmutant, this should be rebase-merged to keep the refactor separate from the fix.

Don't worry, I'll default to rebase-merge all your PRs. I don't like squashing in general, but sometimes PR commit history is just too hairy for a rebase.

**Comment by pgaskin** [comment]
Okay, thanks :wink:. Probably do the same for @gavtroy's as well since those are also logically split into independent commits.

**Pull request merged**